app/presenters/tree_builder.rb

```diff
@@ -404,25 +426,28 @@ def x_get_tree_objects(parent, options, count_only, parents)
     children_or_count || (count_only ? 0 : [])
   end
 
-  # Return a tree node for the passed in object
-  def x_build_node(object, pid, options)    # Called with object, tree node parent id, tree options
+  # @param object the current node object (or an ancestry tree hash)
+  # @param pid [String|Nil] parent id root nodes are nil
+  # @param options [Hash] tree options
+  # @returns [Hash] display hash for this node and all children
+  def x_build_node(object, pid, options)
     parents = pid.to_s.split('_')
 
     options[:is_current] =
         ((object.kind_of?(MiqServer) && MiqServer.my_server(true).id == object.id) ||
          (object.kind_of?(Zone) && MiqServer.my_server(true).my_zone == object.name))
 
+    object, ancestry_kids = object_from_ancestry(object)
     node = x_build_single_node(object, pid, options)
 
     # Process the node's children
-    if Array(@tree_state.x_tree(@name)[:open_nodes]).include?(node[:key]) ||
-       options[:open_all] ||
+    node[:expand] = Array(@tree_state.x_tree(@name)[:open_nodes]).include?(node[:key]) || !!options[:open_all] || node[:expand]
```

Comment:
note to future self:
the changes here are from #9071 which was originally merged first.
It was deemed to be safer if we included those code changes for this line rather than go back to pure darga here.

Also, we deemed #9071 unnecessary at this time on darga.